src/bin/diff.js

```diff
@@ -2,95 +2,107 @@
 
 'use strict';
 
-const path = require('path');
-const argv = require('yargs')
-  .boolean('verbose')
-  .alias('v', 'verbose')
-  .argv;
-
-const cli = require('../lib/cli-util');
-
-const getRuleFinder = require('../lib/rule-finder');
-const arrayDifference = require('../lib/array-diff');
-const objectDifference = require('../lib/object-diff');
-const getSortedRules = require('../lib/sort-rules');
-const flattenRulesDiff = require('../lib/flatten-rules-diff');
-const stringifyRuleConfig = require('../lib/stringify-rule-config');
-
-const files = [argv._[0], argv._[1]];
-const collectedRules = getFilesToCompare(files).map(compareConfigs);
-
-const rulesCount = collectedRules.reduce(
-  (prev, curr) => {
-    return prev + (curr && curr.rules ? curr.rules.length : /* istanbul ignore next */ 0);
-  }, 0);
-
-/* istanbul ignore else */
-if (argv.verbose || rulesCount) {
-  cli.push('\ndiff rules\n' + rulesCount + ' rules differ\n');
-}
-
-/* istanbul ignore else */
-if (rulesCount) {
-  collectedRules.forEach(diff => {
-    let rules = diff.rules;
+const checkEslint = require('../lib/check-eslint');
 
-    /* istanbul ignore if */
-    if (rules.length < 1) {
-      return;
-    }
+if (checkEslint.check() === false) {
```

Comment:
couldn't we just `require('eslint')` and make it a peer dep?

Reply:
It currently _is_ a peer dep and we just require it in `lib/rule-finder`, but in my experience peer deps are not installed automatically so I thought it would be good to have a nice warning.
Eventually I want to avoid confusion as in #285

Reply:
... and instead of just requiring eslint (that is, in a try-catch, to check for it), this draft of an implementation also pulls the corresponding information from the package.json and tests if the installed version satisfies it.

On a second thought, that might be a good idea to even make it a separate, general purpose module and just use that (and concentrate on core functionality of this tool).  
What do you think?

Reply:
I think that letting the require throw naturally, and expecting users to run `npm ls` themselves, is fine - i don’t see much value in an explicit check.

Reply:
Ok. I have to admit, I didn‘t realize this check would be as bloated as it got already (and there still is erroneously assumed, the package.json to be tested again would be found at process.cwd, so it would be even more bloated, if done properly).
I‘ll happily close this then.

But to hijack this discussion a little bit: What about a proper usage information (using the functionality provided by yargs)?

Reply:
That part sounds awesome; I love yargs and its auto-help feature.